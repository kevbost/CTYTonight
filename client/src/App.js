import React from 'react'
import autoBind from 'react-autobind'
import axios from 'axios'
import logo from './logo.svg'
import './App.css'

class App extends React.Component {
  constructor( props ){
    super( props )
    this.state = {
      response: ''
    }
    autoBind( this )
  }

  componentDidMount() {
    this.callApi()
      .then( res => {
        return this.setState({
          response: res.data.express
        })
      })
      .catch( err => { return console.log( err ) }) // eslint-disable-line
  }

  callApi() {
    return axios.get( '/api/hello' )
      .then( response => {
        return response
      })
      .catch( err => {
        console.log( err ) // eslint-disable-line
      })
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">{'Welcome to React'}</h1>
        </header>
        <p className="App-intro">{this.state.response}</p>
      </div>
    )
  }
}

export default App
