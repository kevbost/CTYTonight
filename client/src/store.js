// REDUX THROWAWAY
import { createStore, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import createHistory from 'history/createBrowserHistory'
import { routerMiddleware } from 'react-router-redux'
import reducer from './reducer'

export const history = createHistory()

const store = createStore(
  reducer,
  compose(
    applyMiddleware(
      routerMiddleware( history ),
      thunkMiddleware
    ),
    window.devToolsExtension ? window.devToolsExtension() : f => { return f }
  )
)

export default store
