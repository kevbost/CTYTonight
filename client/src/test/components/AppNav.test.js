import React from 'react'
import App from '../../views/app'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import store, { history } from '../../store'
import renderer from 'react-test-renderer'

const target = document.querySelector( '#root' )

const Wrapper = () => {
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <App />
      </ConnectedRouter>
    </Provider>,
    target
  )
}

describe( 'core tests', () => {
  test( 'renders without crashing', () => {
    const tree = renderer
      .create( <Wrapper /> )
      .toJSON()
    expect( tree ).toMatchSnapshot()
  })
})
