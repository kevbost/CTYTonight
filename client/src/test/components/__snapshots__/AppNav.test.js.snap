// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core tests renders without crashing 1`] = `null`;
